exporter/signalfxexporter/dpclient.go

```diff
@@ -90,13 +119,46 @@ func (s *sfxDPClient) pushMetricsData(
 			"HTTP %d %q",
 			resp.StatusCode,
 			http.StatusText(resp.StatusCode))
-		return exporterhelper.NumTimeSeries(md), err
+		return numTimeseries, err
 	}
 
 	return numDroppedTimeseries, nil
+
+}
+
+func timeseriesCount(metricsData []consumerdata.MetricsData) int {
+	numTimeseries := 0
+	for _, metricData := range metricsData {
+		numTimeseries += exporterhelper.NumTimeSeries(metricData)
+	}
+	return numTimeseries
 }
 
-func buildHeaders(config *Config) (map[string]string, error) {
+func (s *sfxDPClient) metricsByAccessToken(md pdata.Metrics) map[string][]consumerdata.MetricsData {
+	metricsData := pdatautil.MetricsToMetricsData(md)
+
+	// If access token pass through is disabled, return.
+	if !s.accessTokenPassthrough {
+		return map[string][]consumerdata.MetricsData{
+			"": metricsData,
+		}
```

Comment:
Map allocation if pretty expensive. I'm not sure if we need it assuming that almost all the time it'll be just one key.

Reply:
What do you think of the following approach?

```
var currentToken string
var batchStartIdx int
for i, md := metricData {
    metricToken := getToken(md)
    if currentToken != metricToken {
       if batchStartIdx < i {
           pushMetricsData(metricData[batchStartIdx:i])
           batchStartIdx = i
       }
       currentToken = metricToken
    }
} 
```
Theoretically it'll may result in more pushes, but should be more performant even for the cases like `traceData{token1, token2, token1, token2}`.

Otherwise I would use slice of pairs instead of map. if should more performant given that we should not have more than 10's tokens per batch.

Reply:
Updated to incorporate the approach you suggested. Giving it some thought, I think on average it should perform better even though the worst case scenario would result in many pushes. Let me know what you think.